swarm.go

```diff
@@ -1,19 +1,74 @@
 package config
 
 type SwarmConfig struct {
-	AddrFilters             []string
+	// AddrFilters specifies a set libp2p addresses that we should never
+	// dial or receive connections from.
+	AddrFilters []string
+
+	// DisableBandwidthMetrics disables recording of bandwidth metrics for a
+	// slight reduction in memory usage. You probably don't need to set this
+	// flag.
 	DisableBandwidthMetrics bool
-	DisableNatPortMap       bool
-	DisableRelay            bool
-	EnableRelayHop          bool
 
-	// autorelay functionality
-	// if true, then the libp2p host will be constructed with autorelay functionality.
+	// DisableNatPortMap turns off NAT port mapping (UPnP, etc.).
+	DisableNatPortMap bool
+
+	// DisableRelay explicitly disables the relay transport.
+	//
+	// Deprecated: This flag is deprecated and is overridden by
+	// `Transports.Relay` if specified.
```

Comment:
The idea here is essentially to save us the pain of a migration right? Seems like a good idea.

Reply:
Yep.

Reply:
I'm also logging an error.